src/core/lib/channel/channel_args.h

```diff
@@ -389,6 +394,10 @@ class ChannelArgs {
 
     std::string ToString() const;
 
+    // TODO(tjagtap) : Make this available in the
+    // GetChannelArgsDebugInfo() function after rebasing.
+    std::string GetSourceLocationString() { return location_; };
```

Comment:
Just return SourceLocation here too... materialize the string only at the last moment